File: Data Structures & Algorithms/combination-target-sum/submission-0.cs
public class Solution {
    private List<List<int>> result = new List<List<int>>();

    public List<List<int>> CombinationSum(int[] nums, int target) {
        var currentList = new List<int>();
        Backtrack(0, currentList, 0, nums, target);

        return result;
    }

    private void Backtrack(int i, List<int> currentList, int sum, int[] nums, int target)
    {
        if (sum == target)
        {
            result.Add(currentList.ToList());
            return;
        }

        if (i >= nums.Length || sum > target)
        {
            return;
        }

        currentList.Add(nums[i]);
        var includedSum = sum + nums[i];
        Backtrack(i, currentList, includedSum, nums, target);

        currentList.Remove(currentList.Last());
        Backtrack(i + 1, currentList, sum, nums, target);
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const dict = {};
        for (let i = 0; i < nums.length; i++)
        {
            let j = dict[(target - nums[i])];
            if (j !== undefined)
            {
                return [i, j]
            }

            dict[nums[i]] = i;
        }
    }
}
